src/tike/view.py

```diff
@@ -58,11 +58,85 @@
 import matplotlib.transforms as transforms
 from matplotlib import collections
 import matplotlib.pyplot as plt
+import matplotlib.colors as mplcolors
+import cv2 as cv
 import numpy as np
 import tike.linalg
 
 logger = logging.getLogger(__name__)
 
+def complexHSV_to_RGB( img0 ):
+    """Takes a M x N complex valued array, represents the phase as hue,
+    magnitude as value, and saturation as all ones in a new M x N x 3
+    array. This is then converted to a RGB colorspace for further plotting 
+    using e.g. imshow() or imsave() from matplotlib.
+
+    Parameters
+    ----------
+    img0 : :py:class:`numpy.array`
+        A M x N complex64 numpy array.
+
+    Returns
+    -------
+    rgb_img : :py:class:`numpy.array`
+        The M x N x 3 array which represents the input complex valued array 
+        in a RGB colorspace.
+    """
+    
+    sz      = img0.shape
+
+    hsv_img = np.zeros( ( *sz, 3 ), 'float32' )
+
+    hsv_img[ :, :, 2 ] = np.abs( img0 )
+    hsv_img[ :, :, 2 ] = hsv_img[ :, :, 2 ] - np.min( hsv_img[ :, :, 2 ] )
+    hsv_img[ :, :, 2 ] = hsv_img[ :, :, 2 ] / np.max( hsv_img[ :, :, 2 ] )
+
+    hsv_img[ :, :, 1 ] = np.ones( sz, 'float32' )
+
+    hsv_img[ :, :, 0 ] = np.angle( img0 ) 
+    hsv_img[ :, :, 0 ] = hsv_img[ :, :, 0 ] - np.min( hsv_img[ :, :, 0 ] )
+    hsv_img[ :, :, 0 ] = hsv_img[ :, :, 0 ] / np.max( hsv_img[ :, :, 0 ] )
+
+    rgb_img = mplcolors.hsv_to_rgb( hsv_img )
+
+    return rgb_img
+    
+
+def resize_complex_image( img0, scale_factor_r = 1, scale_factor_c = 1, interpolation = cv.INTER_LINEAR ):
```

Comment:
The scale factors should not be labeled as "r" "c" because that is confusing when the image has "real" and "complex" values. Instead, the scale_factor should be 2-tuple?

Reply:
Just did a new git push to address this...